packages/rollup-contracts/contracts/FullStateManager.sol

```diff
@@ -3,17 +3,38 @@ pragma experimental ABIEncoderV2;
 
 /* Internal Imports */
 import {StateManager} from "./StateManager.sol";
+import {SafetyChecker} from "./SafetyChecker.sol";
 
 /**
  * @title FullStateManager
  * @notice The FullStateManager is used for off-chain tx evaluation. It holds a complete mapping
  *         of all chain storage.
  */
 contract FullStateManager is StateManager {
+    // Add Safety Checker contract
+    SafetyChecker safetyChecker;
+    // for testing: if true, then do not perform safety checking on init code or deployed bytecode
+    bool overrideSafetyChecker;
+
+    address ZERO_ADDRESS = 0x0000000000000000000000000000000000000000;
+
+    // bitwise right shift 28 * 8 bits so the 4 method ID bytes are in the right-most bytes
     mapping(address=>mapping(bytes32=>bytes32)) ovmContractStorage;
     mapping(address=>uint) ovmContractNonces;
     mapping(address=>address) ovmCodeContracts;
 
+    /**
+     * @notice Construct a new FullStateManager with a specified safety checker.
+     * @param _opcodeWhitelistMask A bit mask representing which opcodes are whitelisted or not for our safety checker
+     * @param _overrideSafetyChecker Set to true to disable safety checking (WARNING: Only do this in test environments)
+     */
+    constructor(uint256 _opcodeWhitelistMask, bool _overrideSafetyChecker) public {
+        // Set override safety checker flag
+        overrideSafetyChecker = _overrideSafetyChecker;
+        // Set the safety checker address -- NOTE: we assume this contract is deployed by the ExecutionManager
```

Comment:
I think this NOTE is outdated as we appear to be deploying it right below.

Reply:
Sorry this is just a bad comment. "this" refers to the FullStateManager not the SafetyChecker -- updating to:
```
        // Set the safety checker address -- NOTE: `msg.sender` is used as EM address because we assume
        // the FullStateManager is deployed by the ExecutionManager
```

Note I'm not going to do this BS for the partial state manager. We really gotta refactor these contracts so they don't deploy their dependences & I'ma make sure to do that.